src/app/resolvers/contingency-test/contingency-test.resolver.ts

```diff
@@ -25,39 +23,32 @@ import {
 } from 'rxjs';
 import { v4 as uuidv4 } from 'uuid';
 
-@Injectable({
-  providedIn: 'root',
-})
-export class ContingencyTestResolver implements Resolve<boolean> {
-  constructor(
-    private store: Store<State>,
-    private action$: Actions,
-    private techRecordService: TechnicalRecordService,
-    private userService: UserService,
-  ) {}
-
-  resolve(): Observable<boolean> {
-    return this.techRecordService.techRecord$.pipe(
-      switchMap((techRecord) => {
-        const { vin, systemNumber } = techRecord as TechRecordType<'get'>;
-        const vrm = techRecord?.techRecord_vehicleType !== 'trl' ? techRecord?.primaryVrm : undefined;
-        const trailerId = techRecord?.techRecord_vehicleType === 'trl' ? techRecord.trailerId : undefined;
-        return this.store.select(selectTechRecord).pipe(
-          withLatestFrom(this.userService.user$),
-          map(([viewableTechRecord, user]) => {
-            const now = new Date();
-            return {
-              vin,
-              vrm,
-              trailerId,
-              systemNumber,
-              vehicleType: viewableTechRecord?.techRecord_vehicleType,
-              statusCode: viewableTechRecord?.techRecord_statusCode,
-              testResultId: uuidv4(),
-              euVehicleCategory: (viewableTechRecord as TechRecordType<'get'>)?.techRecord_euVehicleCategory ?? null,
-              vehicleSize: viewableTechRecord?.techRecord_vehicleType === 'psv' ? viewableTechRecord?.techRecord_vehicleSize : undefined,
-              vehicleConfiguration: (viewableTechRecord as TechRecordType<'get'>)?.techRecord_vehicleConfiguration ?? null,
-              vehicleClass:
+export const contingencyTestResolver: ResolveFn<boolean> = (route: ActivatedRouteSnapshot, state: RouterStateSnapshot) => {
+  const store: Store<State> = inject(Store<State>);
```

Comment:
Not really an issue, just a stylistic thing. You don't need explicitly type variables as their type will be inferred, and doing so makes the code a bit cleaner and easier to read.